### DISCUSSION TITLE
How Prisma should work with microservice architecture ?

### DISCUSSION THREAD

**Discussion by MohannadBahaa** [prisma/prisma#6243] [Q&A]
### Hey there <img src="https://media.giphy.com/media/hvRJCLFzcasrR4ia7z/giphy.gif" width="25px">

I want to centralized Prisma for my microservices, so I will be able to deal with the latest generated Prisma client as well as I need to migrate/introspect from one place.

Note: I am using a shared database between microservices.

**_for example:_**

<img width="391" alt="Screen Shot 2021-03-25 at 1 35 28 PM" src="https://user-images.githubusercontent.com/28856488/112466851-f9b15700-8d6e-11eb-933b-a794f5f35bf4.png">

  

I have some approach in my mind such as **serverless** or **separate npm package**.

**Comment by ryands17** [comment]
@MohannadBahaa 👋 

A separate package (like an NPM package) would work best in this case as you can share Prisma Client with multiple micro services and can manage the schema and API from a single package.

**Reply by MohannadBahaa** [reply]
thanks  @ryands17

**Reply by MohannadBahaa** [reply]
@ryands17  as I mentioned above it is one of the solutions but I am not sure about the downside here?

I was looking for an example for a couple of days but I could not find a well maintained one.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
